arch/risc-v/include/syscall.h

```diff
@@ -207,7 +164,15 @@ static inline uintptr_t sys_call0(unsigned int nbr)
 
   asm volatile
     (
+#if defined (CONFIG_ARCH_USE_S_MODE) && defined (__KERNEL__)
```

Comment:
This should now select the correct implementation per referrer, so for kernel in S-mode the trampoline is used, and for flat/protected/user mode in kernel build, ecall is used